client/src/App.js

```diff
@@ -22,16 +23,36 @@ function App() {
 
   useEffect(() => {
     userGet()(dispatch);
+    return () => dispatch("FINISH_LOAD");
   }, [dispatch]);
-  console.log(state);
+  console.log((() => !state?.user?.id)());
 
   return (
     <userContext.Provider value={state.user}>
       <MuiThemeProvider theme={theme}>
         <BrowserRouter>
+          <ProtectedElement
+            // ! must remain within Router
+            // ? still don't like the location condition, but unsure of how to handle index
+            condition={() => state.user.id && window.location.pathname !== "/"}
```

Comment:
should we be using react router window / location (one of the react router hooks https://reactrouter.com/web/api/Hooks) in case this changes and we need to re-render this properly. 

Also, I think the navbar shouldn't show for the onboarding page too? So in that case, we might need more conditions :( Or we can have a structure that is more nested React components, and have a dashboard component and then put the navbar and all of those other routes in the dashboard in a switch statement inside the dashboard component

Reply:
I anticipated removing that condition entirely, so getting the location from the window is more a throwaway/open question.

Definitely feel like a wrapper is the direction to move.